src/example/prompt.service.ts

```diff
@@ -0,0 +1,38 @@
+import { Injectable } from '@nestjs/common';
+import { HttpService } from '@nestjs/axios';
+import { Observable } from 'rxjs';
+import { map } from 'rxjs/operators';
+
+@Injectable()
+export class PullRequestReviewService {
+  constructor(private readonly httpService: HttpService) {}
+
+  async getPullRequestDetails(repo: string, prId: number): Promise<any> {
+    const url = `https://api.github.com/repos/${repo}/pulls/${prId}`;
+    return this.httpService.get(url).pipe(map(response => response.data)).toPromise();
+  }
+
+  async reviewPullRequest(repo: string, prId: number): Promise<void> {
+    const details = await this.getPullRequestDetails(repo, prId);
+
+    if (details.changed_files > 10) {
+      this.addComment(repo, prId, "Too many files changed in this PR.");
+    }
+
+    if (!details.title.includes("JIRA-")) {
+      this.addComment(repo, prId, "PR title does not include JIRA ticket.");
+    }
+
+    this.mergePullRequest(repo, prId);
+  }
```

Comment:
PullProofAI

The `reviewPullRequest` method seems to be doing multiple tasks (checking file changes, title validation, and merging). It's recommended to refactor this method into smaller, more focused methods to adhere to the single responsibility principle.
```diff
+   async reviewPullRequest(repo: string, prId: number): Promise<void> {
+     const details = await this.getPullRequestDetails(repo, prId);
+     
+     this.checkFileChanges(repo, prId, details);
+     this.validateTitle(repo, prId, details);
+     this.mergePullRequest(repo, prId);
+   }
+
+   private checkFileChanges(repo: string, prId: number, details: any): void {
+     if (details.changed_files > 10) {
+       this.addComment(repo, prId, "Too many files changed in this PR.");
+     }
+   }
+
+   private validateTitle(repo: string, prId: number, details: any): void {
+     if (!details.title.includes("JIRA-")) {
+       this.addComment(repo, prId, "PR title does not include JIRA ticket.");
+     }
+   }
```



<!-- This is an auto-generated comment by PullProofAI -->